test/components/converters/test_json.py

```diff
@@ -0,0 +1,385 @@
+        "laureates": [
+            {
+                "id": "674",
+                "firstname": "Dario",
+                "surname": "Fo",
+                "motivation": "who emulates the jesters of the Middle Ages in scourging authority and upholding the dignity of the downtrodden",
+                "share": "1",
+            }
+        ],
+    },
+    {
+        "year": "1986",
+        "category": "medicine",
+        "laureates": [
+            {
+                "id": "434",
+                "firstname": "Stanley",
+                "surname": "Cohen",
+                "motivation": "for their discoveries of growth factors",
+                "share": "2",
+            },
+            {
+                "id": "435",
+                "firstname": "Rita",
+                "surname": "Levi-Montalcini",
+                "motivation": "for their discoveries of growth factors",
+                "share": "2",
+            },
+        ],
+    },
+    {
+        "year": "1938",
+        "category": "physics",
+        "laureates": [
+            {
+                "id": "46",
+                "firstname": "Enrico",
+                "surname": "Fermi",
+                "motivation": "for his demonstrations of the existence of new radioactive elements produced by neutron irradiation, and for his related discovery of nuclear reactions brought about by slow neutrons",
+                "share": "1",
+            }
+        ],
+    },
+]
+
+
+def test_init_without_jq_schema_and_content_key():
+    with pytest.raises(
+        ValueError, match="No `jq_schema` nor `content_key` specified. Set either or both to extract data."
+    ):
+        JSONConverter()
+
+
+@patch("haystack.components.converters.json.jq_import")
+def test_init_without_jq_schema_and_missing_dependency(jq_import):
+    converter = JSONConverter(content_key="foo")
+    jq_import.check.assert_not_called()
+    assert converter._jq_schema is None
+    assert converter._content_key == "foo"
+    assert converter._meta_fields is None
+
+
+@patch("haystack.components.converters.json.jq_import")
+def test_init_with_jq_schema_and_missing_dependency(jq_import):
+    jq_import.check.side_effect = ImportError
+    with pytest.raises(ImportError):
+        JSONConverter(jq_schema=".laureates[].motivation")
+
+
+def test_init_with_jq_schema():
+    converter = JSONConverter(jq_schema=".")
+    assert converter._jq_schema == "."
+    assert converter._content_key is None
+    assert converter._meta_fields is None
+
+
+def test_to_dict():
+    converter = JSONConverter(
+        jq_schema=".laureates[]", content_key="motivation", extra_meta_fields=["firstname", "surname"]
+    )
+
+    assert converter.to_dict() == {
+        "type": "haystack.components.converters.json.JSONConverter",
+        "init_parameters": {
+            "content_key": "motivation",
+            "jq_schema": ".laureates[]",
+            "extra_meta_fields": ["firstname", "surname"],
+        },
+    }
+
+
+def test_from_dict():
```

Comment:
Let's also add a test to check the serialization roundtrip (pipeline -> YAML -> pipeline).

Reply:
In several occasions, we've discovered issues of certain components being unserializable to YAML due to unsupported types, etc. Let's add it here, but I think we can probably create a single test to automate this for all components/integrations. I'll create a tracking issue.

Reply:
> I think we can probably create a single test to automate this for all components/integrations. I'll create a tracking issue.

That would be much better in my opinion.